cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,242 @@
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return nil
+}
+
+type TransactionInfo struct {
+	Result           []byte `json:"result"`
+	Meta             []byte `json:"meta"`
+	Envelope         []byte `json:"envelope"`
+	FeeBump          bool   `json:"feeBump"`
+	ApplicationOrder int32  `json:"applicationOrder"`
+	Successful       bool   `json:"successful"`
+	LedgerSequence   uint   `json:"ledgerSequence"`
+}
+
+type GetTransactionsResponse struct {
+	Transactions               []TransactionInfo              `json:"transactions"`
+	LatestLedger               int64                          `json:"latestLedger"`
+	LatestLedgerCloseTimestamp int64                          `json:"latestLedgerCloseTimestamp"`
+	Pagination                 *TransactionsPaginationOptions `json:"pagination"`
+}
+
+type transactionsRPCHandler struct {
+	ledgerReader      db.LedgerReader
+	ledgerEntryReader db.LedgerEntryReader
+	maxLimit          uint
+	defaultLimit      uint
+	logger            *log.Entry
+	networkPassphrase string
+}
+
+func (h *transactionsRPCHandler) getTransactionsByLedgerSequence(ctx context.Context, request GetTransactionsRequest) (GetTransactionsResponse, error) {
+	err := request.isValid(h.maxLimit)
+	if err != nil {
+		return GetTransactionsResponse{}, &jrpc2.Error{
+			Code:    jrpc2.InvalidParams,
+			Message: err.Error(),
+		}
+	}
+
+	// Move start to pagination cursor
+	start := transactions.NewCursor(request.StartLedger, 0, 0)
+	limit := h.defaultLimit
+	if request.Pagination != nil {
+		if request.Pagination.Cursor != nil {
+			start = *request.Pagination.Cursor
+			// increment tx index because, when paginating,
+			// we start with the item right after the cursor
+			start.TxIdx++
+		}
+		if request.Pagination.Limit > 0 {
+			limit = request.Pagination.Limit
+		}
+	}
+
+	// Iterate through each ledger and its transactions until limit or end range is reached
+	var txns []TransactionInfo
+	var cursor transactions.Cursor
+LedgerLoop:
+	for ledgerSeq := start.LedgerSequence; ledgerSeq <= request.EndLedger; ledgerSeq++ {
+		// Get ledger close meta from db
+		ledger, found, err := h.ledgerReader.GetLedger(ctx, ledgerSeq)
+		if (err != nil) || (!found) {
+			if err == nil {
+				err = errors.New("ledger close meta not found")
+			}
+			return GetTransactionsResponse{}, &jrpc2.Error{
+				Code:    jrpc2.InvalidParams,
+				Message: err.Error(),
+			}
+		}
+
+		// Initialise tx reader.
+		reader, err := ingest.NewLedgerTransactionReaderFromLedgerCloseMeta(h.networkPassphrase, ledger)
+		if err != nil {
+			return GetTransactionsResponse{}, &jrpc2.Error{
+				Code:    jrpc2.InvalidParams,
+				Message: err.Error(),
+			}
+		}
+
+		// Move the reader to specific tx idx
+		startTxIdx := 0
+		if ledgerSeq == start.LedgerSequence {
+			startTxIdx = int(start.TxIdx)
+		}
+		err = reader.Seek(startTxIdx)
+		if err != nil {
+			// Seek returns EOF so we can move onto next ledger
+			continue
+		}
```

Comment:
I know Seek only ever returns `io.EOF || nil` but an explicit check would be defensive here